File: mobile/android/app/src/main/java/com/volttracker/obdpoc/data/ObdSessionRecord.java
package com.volttracker.obdpoc.data;

public final class ObdSessionRecord {
    public final long id;
    public final String mode;
    public final String adapterAddress;
    public final String adapterName;
    public final long startedAtMs;
    public final long endedAtMs;
    public final String status;
    public final String supportedPids;
    public final int sampleCount;
    public final long lastEventAtMs;

    ObdSessionRecord(
            long id,
            String mode,
            String adapterAddress,
            String adapterName,
            long startedAtMs,
            long endedAtMs,
            String status,
            String supportedPids,
            int sampleCount,
            long lastEventAtMs) {
        this.id = id;
        this.mode = nonNull(mode);
        this.adapterAddress = nonNull(adapterAddress);
        this.adapterName = nonNull(adapterName);
        this.startedAtMs = startedAtMs;
        this.endedAtMs = endedAtMs;
        this.status = nonNull(status);
        this.supportedPids = nonNull(supportedPids);
        this.sampleCount = sampleCount;
        this.lastEventAtMs = lastEventAtMs;
    }

    private static String nonNull(String value) {
        return value == null ? "" : value;
    }
}

File: mobile/android/app/src/main/kotlin/com/volttracker/obdpoc/data/ObdSessionRecord.kt
package com.volttracker.obdpoc.data

class ObdSessionRecord(
    @JvmField val id: Long,
    mode: String?,
    adapterAddress: String?,
    adapterName: String?,
    @JvmField val startedAtMs: Long,
    @JvmField val endedAtMs: Long,
    status: String?,
    supportedPids: String?,
    @JvmField val sampleCount: Int,
    @JvmField val lastEventAtMs: Long,
) {
    @JvmField val mode: String = mode ?: ""

    @JvmField val adapterAddress: String = adapterAddress ?: ""

    @JvmField val adapterName: String = adapterName ?: ""

    @JvmField val status: String = status ?: ""

    @JvmField val supportedPids: String = supportedPids ?: ""
}
